src/AirshipBindings.iOS.Core/Additions/UAirship.cs

```diff
@@ -0,0 +1,72 @@
+﻿/*
+ Copyright Airship and Contributors
+*/
+
+using System;
+using System.Reflection;
+using Foundation;
+using ObjCRuntime;
+
+namespace UrbanAirship
+{
+    public partial class UAirship
+    {
+        static UAirship()
```

Comment:
You should be able to do this exact thing on Android. Use the shared with the onready callback